custom/icds/messaging/indicators.py

```diff
@@ -238,8 +240,13 @@ class AWWSubmissionPerformanceIndicator(AWWIndicator):
     def __init__(self, domain, user):
         super(AWWSubmissionPerformanceIndicator, self).__init__(domain, user)
 
-        result = get_last_submission_time_for_users(self.domain, [self.user.get_id], self.get_datespan())
-        self.last_submission_date = result.get(self.user.get_id)
+        span = self.get_datespan()
+        self.last_submission_date = ApplicationStatusFact.objects.filter(
+            user_dim__user_id=user.get_id,
+            domain=domain,
+            last_form_submission_date__gte=span.startdate.date(),
+            last_form_submission_date__lte=span.enddate.date(),
```

Comment:
we can probably reduce the number of filters and remove the filters on submission date if we update the code in get_messages() to calculate whether it's been more than a month. i think the datespan filtering was there before to help es, but here we know we're matching basically only 1 record so it shouldn't be necessary.

also we'll need to update the code in get_messages() below to just use self.now in the subtraction because right now it's trying to subtract a date and a datetime.

Reply:
https://github.com/dimagi/commcare-hq/pull/21581/commits/51aa22af0f880115d17e0df9da2ce81f4d21c415